blog/utils/handlePosts.ts

```diff
@@ -96,10 +115,11 @@ export const filterPosts = (
 export default function handlePosts(
   posts: BlogPost[],
   sortBy: SortBy,
-  slug?: string,
+  slug?: string | string[],
   term?: string,
+  excludePostSlug?: string,
 ) {
-  const filteredPosts = filterPosts(posts, slug, term);
+  const filteredPosts = filterPosts(posts, slug, term, excludePostSlug);
```

Comment:
`excludePostSlug` is only being applied to `filterRelatedPosts`.

Filter this post slug from any result:

```suggestion
  const filteredPosts = filterPosts(posts, slug, term).filter(
      ({ slug: postSlug }) => postSlug !== excludePostSlug)
   );
```